src/measure_theory/interval_integral.lean

```diff
@@ -394,6 +394,20 @@ begin
     preimage_neg_Ioc, neg_neg, restrict_congr_set Ico_ae_eq_Ioc]
 end
 
+/-- If two functions are equal in the relevant interval, their interval integrals are also equal. -/
+lemma integral_congr {a b : ℝ} (f g : ℝ → E) (h : ∀ x ∈ interval a b, f x = g x) :
+  ∫ x in a..b, f x = ∫ x in a..b, g x :=
+begin
+  cases le_total a b with hab hab,
+  { simpa only [interval_integral, hab, Ioc_eq_empty, measure.restrict_empty, min_eq_left,
+                max_eq_right, integral_zero_measure, sub_zero]
+      using set_integral_congr is_measurable_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx)) },
+  { simpa only [interval_integral, interval, hab, Ioc_eq_empty, measure.restrict_empty,
+                min_eq_right, max_eq_left, integral_zero_measure, sub_zero, zero_sub,
+                eq_neg_iff_eq_neg, neg_neg]
+      using (set_integral_congr is_measurable_Ioc (λ x hx, h x (mem_Icc_of_Ioc hx))).symm },
```

Comment:
Just style, but I would tend to streamline the `simp` list by using `simp` rather than `simp only`.  Also, can you see if it is possible to change `h` to `(h : eq_on f g (interval a b))` without breaking anything later?  (They are definitionally equal so hopefully this will work.). If so, it's possible to express the `(λ x hx, h x (mem_Icc_of_Ioc hx))` by standard lemmas, as `(h.mono Ioc_subset_Icc_self)`.  Thus:
```suggestion
  { simpa [integral_of_le, hab]
      using set_integral_congr is_measurable_Ioc (h.mono Ioc_subset_Icc_self) },
  { simpa [integral_of_ge, hab]
      using set_integral_congr is_measurable_Ioc (h.mono Ioc_subset_Icc_self) },
```

Reply:
I am about to push a new version with this change, but instead I did

```
  all_goals { simpa [hab, integral_of_le, integral_of_ge]
    using set_integral_congr is_measurable_Ioc (h.mono Ioc_subset_Icc_self) }
```
Do you approve? I don't know if this tactic is allowed in mathlib...

Reply:
I think in that case you could even remove the `begin` and `end` and the following should work:
```lean
by cases le_total a b with hab hab; simpa [hab, integral_of_le, integral_of_ge]
    using set_integral_congr is_measurable_Ioc (h.mono Ioc_subset_Icc_self)

```

Reply:
It does! I pushed the changes.

Reply:
Awesome!